LOGGING_STANDARDS.md

```diff
@@ -1,58 +1,150 @@
+
+| Field name   | Required       | Type                       | Example                      | Description
+| ------------ | -------------- | -------------------------- | ---------------------------- | -----------------------------
+| `created_at` | Yes            | `datetime`<sup>1</sup>     | `"2019-01-21T16:19:12.356Z"` | Date and time of the log event (ISO8601 extended date and time string)
+| `namespace`  | Yes            | `string`                   | `"dp-frontend-router"`       | Service name or other identifier
+| `event`      | Yes            | `string`                   | `"connecting to mongodb"`    | [The event being logged](#effective-event-types)
+| `trace_id`   | No<sup>2</sup> | `string`                   |                              | [Trace ID from OpenCensus](https://opencensus.io/tracing/span/traceid/)
+| `span_id`    | No             | `string`                   |                              | [Span ID from OpenCensus](https://opencensus.io/tracing/span/spanid/)
+| `severity`   | No             | `int8`                     | `2`                          | [The event severity level code](#severity-levels)
+| `http`       | No             | [`http`](#http-event-data) |                              | [HTTP event data](#http-event-data)
+| `auth`       | No             | [`auth`](#auth-event-data) |                              | [Authentication event data](#auth-event-data)
+| `data`       | No             | `object`<sup>3</sup>       |                              | Arbitrary key-value pairs (see below for a full explanation)
+
+<sup>1</sup> All dates must be UTC and in ISO8601 extended date time format: `yyyy-MM-dd'T'HH:mm:ss.SSSZZ` (e.g. `2019-01-21T16:19:12.356Z`)
+
+<sup>2</sup> Not mandatory, but must be included for all events created during the handling of a request.
+
+<sup>3</sup> The event-specific details in `data` are input as an object/map in code, but are stored as a text string by the centralised logging service to allow additional detail to be added to an event without needing to worry about key name collision.  These details can still be searched using a general text search on the `data` field.
+
+#### HTTP event data
+
+HTTP events are so common that we've defined a specific top-level field to capture it. The `http` field defines the common http event data.
+
+This can be used to log inbound or outbound HTTP event data.
+
+| Field name    | Required | Type                   | Example                      | Description
+| ------------- | -------- | ---------------------- | ---------------------------- | -----------
+| `method`      | Yes      | `string`               | `"GET"`                      | The HTTP method used
+| `path`        | Yes      | `string`               | `"/healthcheck"`             | The request path from the URL
+| `query`       | Yes      | `string`               | `"x=1&y=1"`                  | The unparsed query string from the URL
```

Comment:
It seems odd to me for query to be mandatory, I would expect more requests without it than with it? 

Though I agree where it exists it should certainly be logged, so I suppose by requiring it we can ensure that. Just seems inconsistent with the top-level making HTTP optional